pageserver/src/metrics.rs

```diff
@@ -558,6 +588,30 @@ pub(crate) static STORAGE_IO_TIME: Lazy<HistogramVec> = Lazy::new(|| {
     .expect("failed to define a metric")
 });
 
+#[derive(Debug)]
+struct StorageIoTime {
+    metrics: [Histogram; StorageIoOperation::COUNT],
+}
+
+impl StorageIoTime {
+    fn new() -> Self {
+        let metrics = std::array::from_fn(|i| {
+            let op = StorageIoOperation::from_repr(i).unwrap();
+            let metric = STORAGE_IO_TIME
+                .get_metric_with_label_values(&[op.as_str()])
+                .unwrap();
+            metric
+        });
+        Self { metrics }
+    }
+
+    pub(crate) fn get(&self, op: StorageIoOperation) -> Histogram {
+        self.metrics[op as usize]
+    }
+}
+
+pub(crate) const STORAGE_IO_TIME_METRIC: StorageIoTime = StorageIoTime::new();
```

Comment:
```suggestion
pub(crate) static STORAGE_IO_TIME_METRIC: Lazy<StorageIoTime> = Lazy::new(|| StorageIoTime::new());
```

Then this should work just fine.

StorageIoTime cannot be const because STORAGE_IO_TIME is not const. `static` is good here.